packages/react-form-hooks/src/hooks/field/field.ts

```diff
@@ -0,0 +1,196 @@
+ *
+ * ```typescript
+ * const title = useField('default title');
+ * return (
+ *   <div>
+ *     <label htmlFor="test-field">
+ *       Test field{' '}
+ *       <input
+ *         id="test-field"
+ *         name="test-field"
+ *         value={field.value}
+ *         onChange={field.onChange}
+ *         onBlur={field.onBlur}
+ *       />
+ *     </label>
+ *     {field.error && <p>{field.error}</p>}
+ *   </div>
+ * );
+ * ```
+ *
+ * If using `@shopify/polaris` or other custom components that support `onChange`, `onBlur`, `value`, and `error` props then
+ * you can accomplish the above more tersely by using the ES6 `spread` (...) operator.
+ *
+ * ```typescript
+ * const title = useField('default title');
+ * return (<TextField label="Title" {...title} />);
+ * ```
+ *
+ * @param config - The default value of the input, or a configuration object specifying both the value and validation config.
+ * @param validationDependencies - An array of values for determining when to regenerate the field's validation callbacks. Any value that is referenced by a validator other than those passed into it should be included.
+ * @returns A `Field` object representing the state of your input. It also includes functions to manipulate that state. Generally, you will want to pass these callbacks down to the component or components representing your input.
+ *
+ * @remarks
+ * **Reinitialization:** If the `value` property of the field configuration changes between calls to `useField`,
+ * the field will be reset to use it as it's new default value.
+ *
+ * **Imperative methods:** The returned `Field` object contains a number of methods used to imperatively alter it's state.
+ * These should only be used as escape hatches where the existing hooks and components do not make your life easy,
+ * or to build new abstractions in the same vein as `useForm`, `useSubmit` and friends.
+ */
+export function useField<Value = string>(
+  input: FieldConfig<Value> | Value,
+  dependencies: unknown[] = [],
+): Field<Value> {
+  const {value, validates} = normalizeFieldConfig(input);
+  const validators = normalizeValidation(validates);
+
+  const [state, dispatch] = useFieldReducer(value);
+
+  const reset = useCallback(() => dispatch(resetAction()), [dispatch]);
+  const newDefaultValue = useCallback(
+    value => dispatch(newDefaultAction(value)),
+    [dispatch],
+  );
+
+  const runValidation = useCallback(
+    (value: Value = state.value) => {
+      const error = validators
+        .map(check => check(value, {}))
+        .filter(value => value != null);
+
+      if (error && error.length > 0) {
+        const [firstError] = error;
+        dispatch(updateErrorAction(firstError));
+        return firstError;
+      }
+
+      dispatch(updateErrorAction(undefined));
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [state.value, ...dependencies],
```

Comment:
Hmm, seems a bit strange to me we'd need to invalidate this on every state change.